scripts/Crawlers/BaseCrawler.py

```diff
@@ -552,3 +558,22 @@ def _check_dats_present(self, directory):
                         return os.path.join(file_path, subfile_name)
             elif file_name.lower() == "dats.json":
                 return file_path
+
+    def _add_source_data_submodule_if_derivedfrom_conp_dataset(
+        self, dats_json, dataset_dir
+    ):
+        f = open(dats_json)
+        metadata = json.loads(f.read())
+        f.close()
+
+        source_dataset_link = None
+        source_dataset_id = None
+        for property in metadata["extraProperties"]:
```

Comment:
I don't think `extraProperties` is part of the required DATS properties. Could there be a case where the `"extraProperties"` key does not exist? I think it'd make sense to check this here.

Reply:
I added an if statement right above to check if the property exists. If it does not exist, then it will simply return from the function as there is no information regarding potential source data.